docs/sorting-algorithms/quick-sort.mdx

```diff
@@ -0,0 +1,247 @@
+    //Left part of the pivot
+    quickSort( a, start, p-1 );
+    //Right part of the pivot
+    quickSort( a, p+1, end);
+
+    return;
+}
+
+int main(){
+
+    int n;
+    cout<<"Enter the number of elements: ";
+    cin>>n;
+
+    int a[n];
+    cout<<endl<<"Enter the elements of the array: ";
+    
+    for(int i= 0; i<n; i++){        //For loop to input n elements into the array
+        cin>>a[i];
+    }
+
+    //Call the quick sort function on the array - quickSort( array_name, start, end);
+    quickSort( a, 0, n-1 );
+
+    //Print the sorted array
+    cout<<endl<<"The sorted array is: ";
+    for(int i= 0; i< n; i++){
+        cout<<a[i]<<" ";
+    }
+
+
+    return 0;
+}
+
+```
+</TabItem>
+
+<TabItem value="py">
+
+```python   
+# Program to implement QuickSort Algorithm in Python
+
+
+def partition(arr, low, high):
+    '''
+    The value of i is initialized to (low-1) since initially first element
+    is swapped by itself
+    Reason: no greater element has been encountered apart from itself
+    '''
+    pivotElement = arr[high]
+    i = (low - 1)
+
+    for j in range(low, high):
+        if arr[j] < pivotElement:
+            i += 1
+            # swap elements arr[i] and arr[j]
+            arr[i], arr[j] = arr[j], arr[i]
+
+    # swap pivot element with element at index=(i + 1) since loop ended,
+    # to obtain LHS of pivot
+    arr[i + 1], arr[high] = arr[high], arr[i + 1]
+
+    return(i + 1)
+
+'''
+This is the calling function that implements QuickSort algorithm, where:
+arr = input array given by user
+low = starting index
+high = ending index
+'''
+
+
+def quickSort(arr, low, high):
+    if low < high:
+
+        # pi is partitioning index, arr[p] is now at right place
+        pi = partition(arr, low, high)
+
+        # Separately sort elements before partition and after partition
+        quickSort(arr, low, pi-1)
+        quickSort(arr, pi+1, high)
+
+# main function
+if __name__ == "__main__":
+    arr = []
+    n = int(input("Enter the number of elements: "))
+    print('Enter the elements of the array: ')
+    for i in range(n):
+      arr.append(int(input()))
+    quickSort(arr, 0, n-1)
+    print('The sorted array is: ')
+    for i in range(n):
+        print(arr[i], end=" ")
+    print()
+    
+```
+</TabItem>
+</Tabs>
+
+# Sample Input and Output
+
+
+#### Input :
+
+```
+Enter the number of elements: 5
+
+Enter the elements of the array: 22 11 44 55 33
+```
+#### Output :
+
+```
+The sorted array is: 11 22 33 44 55
+```
+
+# Time & Space Complexity
+ 
+Quick sort algorithm partition the array about pivot. So the worst case is if we pick the pivot element greatest or smallest.
+Because then we have to call partititon function n times on n elements so time complexity becomes: O(n^2).
+And the best case is when we pick the pivot as middle element.So we have to call partititon function logn times
+on n elements. So time complexity becomes: O(nlogn). And in average case also time complexity:O(nlogn).
```

Comment:
Wrap all of this in one line. Also if you have already shown the values below, no need to repeat them here.